src/renderer/src/routes/settings/(menu)/[slug]/components/AppSettings.svelte

```diff
@@ -33,5 +132,145 @@
+							? $appPortQuery.data
+							: $appPortQuery.isLoading
+								? 'loading...'
+								: 'unknown'}
+					</div>
+				</div>
+				<hr class="divider mb-4 mt-4 opacity-60" />
+
+				<div class="flex-end flex flex-1 flex-row items-center">
+					<span class="flex-1"
+						>{$i18n.t('Generate a one-time, expiring authentication token.')}</span
+					>
+					<Button
+						props={{
+							class: 'btn-secondary flex-1 text-al max-w-28',
+							onClick: () => issueAuthToken(300, true),
+							disabled: $issueAuthTokenMutation.isPending
+						}}
+					>
+						<div class="flex flex-row items-center">
+							<span class="font-extrabold">+</span>
+							<span class="ml-1">{$i18n.t('Generate')}</span>
+						</div>
+					</Button>
+				</div>
+				<div class="flex-end flex flex-1 flex-row items-center">
+					<span class="flex-1"
+						>{$i18n.t(
+							"Generate an unlimited use authentication token that's valid until Launcher is quit."
+						)}</span
+					>
+					<Button
+						props={{
+							class: 'btn-secondary flex-1 text-al max-w-28',
+							onClick: () => issueAuthToken(0, false),
+							disabled: $issueAuthTokenMutation.isPending
+						}}
+					>
+						<div class="flex flex-row items-center">
+							<span class="font-extrabold">+</span>
+							<span class="ml-1">{$i18n.t('Generate')}</span>
+						</div>
+					</Button>
+				</div>
+				{#each authTokenResponses.sort((r1, r2) => {
+					if (r1.expires_at && r2.expires_at) return r2.expires_at - r1.expires_at;
+					if (r1.expires_at) return -1;
+					if (r2.expires_at) return 1;
+					return 0;
+				}) as response}
+					<div class="column card mb-2 mb-2 items-center p-3 text-base dark:variant-soft-tertiary">
+						{#if response.expires_at}
+							<div class="row flex-end flex text-sm opacity-60">
+								<span class="flex-1"></span>
+								{$i18n.t('expires')}
+								{new Date(response.expires_at / 1000)}
+							</div>
+						{/if}
+						<button
+							class="mt-1 break-all text-left hover:text-white"
+							title={$i18n.t('Click to Copy')}
+							on:click={() => {
+								navigator.clipboard.writeText(
+									Base64.fromUint8Array(new Uint8Array(response.token))
+								);
+								toastStore.trigger({
+									message: `Copied.`
+								});
+							}}
+						>
+							{Base64.fromUint8Array(new Uint8Array(response.token))}
+						</button>
+					</div>
+				{/each}
+				<hr class="divider mb-4 mt-4 opacity-60" />
+				<div class="flex flex-1 flex-col">
+					<div class="mb-4 flex-1 font-bold">
+						{$i18n.t('Grant permission to someone to take actions on your behalf:')}
+					</div>
+					<div class="mb-4">
+						<InputWithLabel
+							bind:value={agentToGrantSigningCredentials}
+							id="pubkeyInput"
+							label={$i18n.t('publicKey')}
+						/>
+					</div>
+					<!-- <div class="mb-4">
+						<InputWithLabel bind:value={capSecret} id="capSecret" label={$i18n.t('secret')} />
+					</div> -->
```

Comment:
Why is this commented out?

Reply:
Good question! It might also be removed. The reason is that a default cap secret will be used now instead of a user provided one. Ant the reason is that cap secrets need to be of fixed length (exactly 64 bytes) and this is not really easily achievable with user input because you can't predict the number of bytes that a string will have.

Reply:
It's not very nice but the main motivation behind this PR was allowing integration of the devhub CLI which is supported with this default cap secret.